### ISSUE TITLE
The correct operator for adding a stride to a slice?

### ISSUE THREAD

**Issue by fsh** [fsh/strides#1]
It's unclear to me whether `@:` is the best operator to use. Feedback is desired along the line of "it's OK", "it's perfect", "it's bad".

I originally wanted to use `@` as I find that operator rather natural for this purpose: `10 .. 20 @ 2`. The cognitive distance between "at 2" and "at every 2nd" is small.

However, this operator is already in use in Nim for the (rather rarefied, in my opinion) purpose of converting strings and arrays to `seq`s. Personally I want to do strided slicing (& reversing) more often than I want to force everything into the performance-cursed `seq` datatype, but I'm not sure if it's fair use.

Alternatives include something like `:/`: `0 .. 10 :/ 2`. Here the cognitive leap is via mathematical intuition. `::` could be considered for the Python feel of it, tho Nim manual states that this one is reserved.

Note that since precedence cannot be specified in Nim, the operator would have to start with one of `=<>!@:?`.

**Comment by AngelEzquerra** [comment]
Arraymancer uses a single | character to indicate strides (e.g. myTensor[0 .. 50 | 2]). I think it’d be nice to use the same syntax. That being said, you might still need a different operator if you want to use it in contexts outside of the access operator (e.g. in for loops).

**Comment by fsh** [comment]
> Arraymancer uses a single | character to indicate strides (e.g. myTensor[0 .. 50 | 2]). I think it’d be nice to use the same syntax.

I agree coherence would be nice but I also feel `|` is problematic for the following reasons:

Due to Nim's (somewhat eccentric) precedence rules `1 .. ^2 | 2` gets parsed as `1 .. (^2 | 2)`. This can be worked around (as Arraymancer does) but it still feels a bit hacky.

`|` is irrevocably associated with `or`/union/join-like operations in my mind. Other associated uses are tiling, concatenation, and in Nim the parallel `||` -- but nothing that really feels intuitive with the concept of striding/stepping.

(One could theoretically imagine there was a sort of `MultiSlice` or something, which could represent any number of subslices -- in such a case I would probably want a `|`-like operation for that instead. E.g. `(0..10) | (20..30)` to mean "`0..30` but excluding the hole `11..19`" Granted, this type would rarely be useful and it would also be very awkward given my first point, again due to inability to specify operator precedence in Nim, so it's very hypothetical.)

For a single-char operator I still think `@` would be the best bet (given that constructions such as `x:y:z` or `x .. y by z` are impossible in Nim).

**Comment by AngelEzquerra** [comment]
In that case I'd say that @, makes the most sense, as "a .. b @ n" could be interpreted as "the elements of a .. b which are found 'at' the nth location". Another alternative would be to use @@ if going for a "symmetry" with the double dot of the slice.
